crates/node/src/commands/setup.rs

```diff
@@ -0,0 +1,176 @@
+use std::path::{Path, PathBuf};
+
+use sc_cli::{CliConfiguration, Error, Result, SharedParams, SubstrateCli};
+use sc_service::BasePath;
+use url::Url;
+
+use crate::chain_spec::GENESIS_ASSETS_DIR;
+use crate::cli::Cli;
+use crate::configs::FileInfos;
+use crate::{configs, constants};
+
+/// Define a way to retrieve an index.json file
+
+/// The index.json must follow the format of the official index.json
+/// (https://github.com/keep-starknet-strange/madara/blob/main/configs/index.json)
+/// Where the `md5` and `url` fields are optional
+#[derive(Debug, clap::Args)]
+#[group(required = true, multiple = false)]
+pub struct SetupSource {
+    /// Download an index.json file for an url
+    #[clap(
+        long,
+        conflicts_with="from_local",
+        value_hint=clap::ValueHint::Url,
+        // This combination of properties allow us to use a default value only if the arg is passed without value.
+        // If it is not passed at all, no default value is used.
+        // See: https://docs.rs/clap/latest/clap/struct.Arg.html#method.default_missing_value
+        num_args = 0..=1, require_equals=true, default_missing_value = Some(constants::DEFAULT_CONFIGS_URL)
+    )]
+    pub from_remote: Option<String>,
+
+    /// Copy an index.json file for an url
+    #[clap(long, conflicts_with = "from_remote")]
+    pub from_local: Option<String>,
+}
+
+#[derive(Debug, clap::Args)]
+pub struct SetupCmd {
+    #[allow(missing_docs)]
+    #[clap(flatten)]
+    pub shared_params: SharedParams,
+
+    #[clap(flatten)]
+    pub source: SetupSource,
+}
+
+impl CliConfiguration for SetupCmd {
+    fn shared_params(&self) -> &SharedParams {
+        &self.shared_params
+    }
+}
+
+impl SetupCmd {
+    pub fn run(&self) -> Result<()> {
+        log::info!("setup cmd: {:?}", self);
+        let dest_config_dir_path = {
+            let is_dev = self.shared_params().is_dev();
+            let chain_id = self.shared_params().chain_id(is_dev);
```

Comment:
`chain_id(true)` ?

Reply:
why that?

Reply:
you can setup for non-dev chains. Local but non dev

Reply:
that is what u are setting there atm, i think
`let is_dev = self.shared_params().is_dev();` = `let is_dev = true;`

Reply:
no, is dev just check if `--dev` was passed
so `madara setup --dev` with have it true and `madara setup --chain=local` will have it false